src/orbits/keplerian.jl

```diff
@@ -0,0 +1,532 @@
+
+    if !isnothing(nt.b)
+        any( (!isnothing).((nt.incl, duration)) ) && throw(ArgumentError(
+            "Only `incl`, `b`, or `duration` can be given"
+        ))
+        b = nt.b
+        cos_incl = dcosi_db * b
+        incl = acos(cos_incl)
+        sin_incl = sin(incl)
+        duration = nt.duration
+    elseif !isnothing(nt.incl)
+        !isnothing(nt.duration) && throw(ArgumentError(
+            "Only `incl`, `b`, or `duration` can be given"
+        ))
+        incl = nt.incl
+        sin_incl, cos_incl = sincos(incl)
+        b = cos_incl / dcosi_db
+        duration = nt.duration
+    elseif !isnothing(nt.duration)
+        duration = nt.duration
+        c = sin(π * duration / (incl_factor_inv) / period)
+        c_sq = c^2
+        ecc_sin_omega = ecc*sin_omega
+        aor = a_planet / R_star
+        b = √(
+            (aor^2 * c_sq - 1.0) /
+            (
+                c_sq * ecc_sin_omega^2 +
+                2.0*c_sq*ecc_sin_omega +
+                c_sq - ecc^4 + 2.0*ecc^2 - 1.0
+            )
+        ) * (1.0 - ecc^2)
+        cos_incl = dcosi_db * b
+        incl = acos(cos_incl)
+        sin_incl = sin(incl)
+    else
+        incl = 0.5*π
+        cos_incl = 0.0
+        sin_incl = 1.0
+        b = 0.0
+        duration = nt.duration
+    end
+
+    # Compute remaining system parameters
+    !(isnothing(nt.t_0) ⊻ isnothing(nt.t_p)) && throw(
+        ArgumentError("Either t₀ or tₚ must be specified")
+    )
+    if isnothing(nt.t_0)
+        t_p = nt.t_p
+        t_0 = t_p + M_0 / n
+    else
+        t_0 = nt.t_0
+        t_p = t_0 - M_0 / n
+    end
+
+    t_ref = t_p - t_0
+
+    # Normalize inputs
+    (
+        period, t_0, t_p, t_ref, duration,
+        a, a_planet, a_star, R_planet, R_star,
+        rho_planet, rho_star,
+        RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega,
+        incl, omega, Omega,
+        M_planet, M_star,
+    ) = normalize_inputs(
+        period, t_0, t_p, t_ref, duration,
+        a, a_planet, a_star, R_planet, R_star,
+        rho_planet, rho_star,
+        RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega,
+        incl, omega, Omega,
+        M_planet, M_star,
+        no_units,
+    )
+    return KeplerianOrbit(
+        period, t_0, t_p, t_ref, duration,
+        a, a_planet, a_star, R_planet, R_star,
+        rho_planet, rho_star,
+        RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega,
+        incl, omega, Omega,
+        n,
+        M_planet, M_star,
+    )
+end
+
+@kwcall KeplerianOrbit(
+    period=nothing, t_0=nothing, t_p=nothing, duration=nothing,
+    a=nothing, R_planet=nothing, R_star=nothing,
+    rho_star=nothing,
+    RpRs=nothing, aR_star=nothing, b=nothing, ecc=nothing, cos_omega=nothing, sin_omega=nothing,
+    incl=nothing, omega=nothing, Omega=nothing,
+    M_planet=nothing, M_star=nothing,
+)
```

Comment:
I've had a couple comments earlier, but I think in general if there are reasonable defaults we should stick those in here. E.g., `incl=0.5\pi`, `omega=0`, `ecc=0`, etc.

Reply:
hmmm, ok. I'm wondering how we would want to go about handling units, e.g., `R_star=1.0` vs. `R_star=1.0u"Rsun"` in `@kwcall`. Or, we could just give default values for the unitless params?

Reply:
Defaults should be unitless, it will always be the users job to get the right unit inputs IMO